### PR TITLE
Format dev error messages

### PR DESCRIPTION
**Pull request by markstory**
Apply a few simple HTML formatting transforms to error messages. This should help make error pages easier to understand and gives us more options when writing verbose error messages.

### CODE REVIEW

File: tests/TestCase/Error/DebuggerTest.php [hunk 1]
```diff
@@ -825,4 +825,24 @@ public function testPrintVar()
         $expected = sprintf($expected, str_replace(CAKE_CORE_INCLUDE_PATH, '', __FILE__), __LINE__ - 9);
         $this->assertEquals($expected, $result);
     }
+
+    /**
+     * test formatHtmlMessage
+     *
+     * @return void
+     */
+    public function formatHtmlMessage()
```
Comment (dereuromark) [inline tests/TestCase/Error/DebuggerTest.php]: Test prefix missing
Tests are not running

File: src/Error/Debugger.php [hunk 2]
```diff
@@ -929,6 +929,29 @@ public static function printVar($var, array $location = [], ?bool $showHtml = nu
...
+     * - Convert `bool` into `<code>bool</code>`
+     * - Convert newlines into `<br />`
+     *
+     * @param string $message The string message to format.
+     * @return string Formatted message.
+     */
+    public static function formatHtmlMessage(string $message): string
+    {
+        $message = h($message);
+        $message = preg_replace_callback('/`([^`]+)`/', function ($matches) {
```
Comment (garas) [inline src/Error/Debugger.php]: `$message = preg_replace(..., '<code>$1</code>', $message);`

Callback unnecessary.

File: templates/layout/dev_error.php [hunk 3]
```diff
@@ -12,14 +12,15 @@
...
+use Cake\Error\Debugger;
 ?>
 <!DOCTYPE html>
 <html>
 <head>
     <?= $this->Html->charset() ?>
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
     <title>
-        Error: <?= h($this->fetch('title')) ?>
+        Error: <?= Debugger::formatHtmlMessage($this->fetch('title')) ?>
```
Comment (garas) [inline templates/layout/dev_error.php]: HTML tags in `<title>` not supported.

File: tests/TestCase/Error/DebuggerTest.php (hunk 1 above)
Comment (markstory) [reply tests/TestCase/Error/DebuggerTest.php]: Whoops :blush:

File: templates/layout/dev_error.php (hunk 3 above)
Comment (markstory) [reply templates/layout/dev_error.php]: Yes, I did. I failed to transpose the change I made in my app skeleton here :blush:

**Pull request merged**